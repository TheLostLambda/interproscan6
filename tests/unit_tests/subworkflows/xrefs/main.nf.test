nextflow_workflow {

    name "Test Workflow XREFS"
    script "interproscan/subworkflows/xrefs/main.nf"
    workflow "XREFS"

    stage {
        symlink "interproscan/modules/xrefs/entries/main.nf"
        symlink "interproscan/modules/xrefs/goterms/main.nf"
        symlink "interproscan/modules/xrefs/paint_annotations/main.nf"
        symlink "interproscan/modules/xrefs/pathways/main.nf"
        copy "tests/unit_tests/test_inputs/xrefs/entries.json"
        copy "tests/unit_tests/test_inputs/xrefs/goterms.json"
        copy "tests/unit_tests/test_inputs/xrefs/goterms.ipr.json"
        copy "tests/unit_tests/test_inputs/xrefs/pathways.json"
        copy "tests/unit_tests/test_inputs/xrefs/pathways.ipr.json"
        copy "interproscan/scripts/xrefs/entries.py"
        copy "interproscan/scripts/xrefs/goterms.py"
        copy "interproscan/scripts/xrefs/paint_annotations.py"
        copy "interproscan/scripts/xrefs/pathways.py"
        copy "tests/unit_tests/test_inputs/xrefs/subworkflow/xref_matches_input.json"
    }

    test("Should run") {

        when {
            params {
            }
            workflow {
                """
                input[0] = file("meta/tests/unit_tests/test_inputs/xrefs/subworkflow/xref_matches_input.json")
                input[1] = "ncbifam,sfld"
                input[2] = "../../../meta/tests/unit_tests/test_inputs"
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
