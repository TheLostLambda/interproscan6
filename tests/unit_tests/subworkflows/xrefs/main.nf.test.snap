{
    "Should run": {
        "content": [
            {
                "0": [
                    "entries_xref_matches_input.json:md5,82e1e06299a896dcfd952ce988bcc935"
                ],
                "final_result": [
                    "entries_xref_matches_input.json:md5,82e1e06299a896dcfd952ce988bcc935"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-09-09T09:30:10.471333"
    }
}